sormas-ui/src/main/java/de/symeda/sormas/ui/environment/EnvironmentController.java

```diff
@@ -15,11 +15,162 @@
 
 package de.symeda.sormas.ui.environment;
 
+import java.util.function.Consumer;
+
 import com.vaadin.navigator.Navigator;
+import com.vaadin.ui.Notification;
+
+import de.symeda.sormas.api.FacadeProvider;
+import de.symeda.sormas.api.caze.InvestigationStatus;
+import de.symeda.sormas.api.common.DeletionReason;
+import de.symeda.sormas.api.deletionconfiguration.DeletionInfoDto;
+import de.symeda.sormas.api.environment.EnvironmentDto;
+import de.symeda.sormas.api.i18n.I18nProperties;
+import de.symeda.sormas.api.i18n.Strings;
+import de.symeda.sormas.api.user.UserRight;
+import de.symeda.sormas.api.utils.DataHelper;
+import de.symeda.sormas.ui.ControllerProvider;
+import de.symeda.sormas.ui.SormasUI;
+import de.symeda.sormas.ui.UserProvider;
+import de.symeda.sormas.ui.utils.ArchiveHandlers;
+import de.symeda.sormas.ui.utils.CommitDiscardWrapperComponent;
+import de.symeda.sormas.ui.utils.VaadinUiUtil;
+import de.symeda.sormas.ui.utils.components.automaticdeletion.DeletionLabel;
+import de.symeda.sormas.ui.utils.components.page.title.TitleLayout;
 
 public class EnvironmentController {
 
 	public void registerViews(Navigator navigator) {
 		navigator.addView(EnvironmentsView.VIEW_NAME, EnvironmentsView.class);
+		navigator.addView(EnvironmentDataView.VIEW_NAME, EnvironmentDataView.class);
+	}
+
+	public void create() {
+		CommitDiscardWrapperComponent<EnvironmentCreateForm> environmentCreateComponent = getEnvironmentCreateComponent();
+		if (environmentCreateComponent != null) {
+			VaadinUiUtil.showModalPopupWindow(environmentCreateComponent, I18nProperties.getString(Strings.headingCreateNewEnvironment));
+		}
+
+	}
+
+	public CommitDiscardWrapperComponent<EnvironmentCreateForm> getEnvironmentCreateComponent() {
+		UserProvider curentUser = UserProvider.getCurrent();
+
+		if (curentUser != null) {
+			EnvironmentCreateForm createForm;
+			createForm = new EnvironmentCreateForm();
+			final EnvironmentDto environment = EnvironmentDto.build();
+			environment.setReportingUser(curentUser.getUserReference());
```

Comment:
please add a `build` method in the `EnvironmentDto` that takes a user and sets these values on the new environment

Reply:
updated